mlflow/tracking/fluent.py

```diff
@@ -3053,32 +3054,53 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
 
 
-_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+@dataclass
+class ActiveModelContext:
+    """
+    The context of the active model.
+
+    Args:
+        model_id: The ID of the active model.
+        set_by_user: Whether the active model was set by the user or not.
+    """
+
+    model_id: Optional[str] = None
+    set_by_user: Optional[bool] = None
+
+
+_ACTIVE_MODEL_CONTEXT = ContextVar(
+    "active_model_context",
+    default=ActiveModelContext(),
+)
```

Comment:
```suggestion
_ACTIVE_MODEL_CONTEXT = ContextVar(
    "active_model_context",
    default=None,
)
```

can we make the default None?

Reply:
Then we need to check if _ACTIVE_MODEL_CONTEXT.get() is None or not before trying to fetch its field, which is more complex.